src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java

```diff
@@ -97,6 +97,6 @@ public String getFieldName() {
 	// DATACOUCH-145: allows SDK's @Id annotation to be used
 	@Override
 	public boolean isIdProperty() {
-		return isAnnotationPresent(Id.class) || super.isIdProperty();
+		return isAnnotationPresent(Id.class) || super.isIdProperty() || this.getFieldName().equals("id");
```

Comment:
maybe check with lower case to be more resilient?

Reply:
done